events_test.go

```diff
@@ -0,0 +1,114 @@
+package slack
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+)
+
+var c = New("my-token")
+
+func TestParserOuterCallBackEvent(t *testing.T) {
+	eventsAPIRawCallbackEvent := `
+			{
+				"token": "XXYYZZ",
```

Comment:
this is the verification token correct? how do we validate these events against the token?

Reply:
I believe these are just shared secrets. Curious if you'd be open to allowing http middleware to do the validation of these and the webhook URL validation. The approach I took in #283 could be complimentary to this PR and do the validation? If you'd rather leave it entirely to the user then I think it's just a string comparison to verify.

Reply:
Hm, might be missing that part @james-lawrence and also thanks @JackWink for the pointers. 

Will re-read some of the Slack docs to ensure correct usage!

Reply:
doing middleware means parsing the json mutliple times, I was mainly pointing out that a function should be included to do a basic comparison so people don't have to write the 4-6 lines themselves =)

Reply:
fair enough, I'd prefer that `ParseEventsAPIEvent` take in the verification token, or allow you to explicitly pass an option not to verify it.  Something like:

```go
type EventApiConfig struct {
    VerificationToken string
    DoNotVerify bool
}

type EventApiOption func(cfg *EventApiConfig)

func UnsafeVerification() EventApiOption {
  return func(cfg *EventApiConfig) {
    cfg.DoNotVerify = true
  }
}

func WithVerificationToken(token string) EventApiOption {
  return func(cfg *EventApiConfig) {
    cfg.VerificationToken = token
  }
}
func ParseEventsApiEvent(msg json.RawMessage, opts ...EventApiOption) {
  cfg := EventApiConfig{}
  for _, opt := opts {
    opt(&cfg)
  }
  ...
  if cfg.DoNotVerify {
    return data, nil
  }

  if cfg.VerificationToken == "" {
    return nil, fmt.Errorf("Missing event verification token")
  }
  
  if subtle.ConstantTimeCompare([]byte(cfg.VerificationToken), []byte(data.Token)) != 1 {
    return nil, fmt.Errorf("Event verification token doesn't match")
  }
  return data, nil
}

Reply:
lets pass on that part atm. I like the idea, except there are cases where we might need to use multiple verification tokens (if you're running multiple bots for example)

Reply:
I think that more advance use-case would fall under the DoNotVerify option, you could still provide the validation function separate from the parser. I feel like if you separate the validation and parsing you're more likely to have users skip validation entirely. IMO secure-by-default with the option to shoot yourself in the foot is better.

Reply:
I mean I'd just make the option take a function that accepts the request's token and returns true/false.

```golang
type verifier interface {
  Verify(t string) bool
}

func OptionVerifyToken(v verifier) Option {
  // ...
}
```

mainly want to avoid bogging down this PR with these details atm.